lintcheck/src/main.rs

```diff
@@ -337,7 +348,9 @@ impl Crate {
         let shared_target_dir = clippy_project_root().join("target/lintcheck/shared_target_dir");
 
         let mut cargo_clippy_args = if config.fix {
-            vec!["--fix", "--"]
+            // need to pass `clippy` arg even if already feeding `cargo-clippy`
+            // see https://github.com/rust-lang/rust-clippy/pull/9461
+            vec!["clippy", "--fix", "--allow-no-vcs", "--"]
         } else {
             vec!["--", "--message-format=json", "--"]
```

Comment:
While you're in the neighbourhood, could change this bit to

```suggestion
            vec!["clippy", "--message-format=json", "--"]
```

I did find this quite confusing before the conversation in https://github.com/rust-lang/rust-clippy/pull/9461